lib/puppet_x/puppetlabs/strings/yard/handlers/host_class_handler.rb

```diff
@@ -8,6 +8,21 @@ class PuppetX::PuppetLabs::Strings::YARD::Handlers::HostClassHandler < PuppetX::
         param_tuple << ( a[1].nil? ? nil : a[1].source )
       end
     end
+    tp = Puppet::Pops::Types::TypeParser.new
+    param_type_info = []
+    # Ensure that this pop object has parameters
+    if statement.pops_obj.respond_to? :parameters
+      param_type_info = {}
+      statement.pops_obj.parameters.each do |pop_param|
+        begin
+          param_type_info[pop_param.name] =  tp.interpret_any(pop_param.type_expr)
+        rescue Puppet::ParseError
+          # If the type could not be interpreted, default to type Any
+          param_type_info[pop_param.name] =  tp.parse('Any')
```

Comment:
No need to create a new instance of 'Any' for each untyped parameter - they can all refer to the same Any instance. To  make this work for all puppet versions call Puppet::Pops::Types::TypeFactory.any() instead (it will provide a singleton in newer puppet versions, and do the right thing in older versions).